junit-jupiter-api/src/main/java/org/junit/jupiter/api/AssertLinesMatch.java

```diff
@@ -0,0 +1,178 @@
+				continue; // main
+			}
+
+			// fast-forward marker found in expected line: fast-forward actual line...
+			if (isFastForwardLine(expectedLine)) {
+				int fastForwardLimit = parseFastForwardLimit(expectedLine);
+
+				// trivial case: fast-forward marker was in last expected line
+				if (expectedDeque.isEmpty()) {
+					int actualRemaining = actualDeque.size();
+					// no limit given or perfect match? we're done.
+					if (fastForwardLimit == Integer.MAX_VALUE || fastForwardLimit == actualRemaining) {
+						return;
+					}
+					fail(expectedLines, actualLines, "terminal fast-forward(%d) error: fast-forward(%d) expected",
+						fastForwardLimit, actualRemaining);
+				}
+
+				// fast-forward limit was given: use it
+				if (fastForwardLimit != Integer.MAX_VALUE) {
+					// fast-forward now: actualDeque.pop(fastForwardLimit)
+					for (int i = 0; i < fastForwardLimit; i++) {
+						actualDeque.pop();
+					}
+					continue; // main
+				}
+
+				// peek next expected line
+				expectedLine = expectedDeque.peek();
+				// fast-forward "unlimited": until next match
+				while (true) {
+					if (actualDeque.isEmpty()) {
+						fail(expectedLines, actualLines, "fast-forward(∞) didn't find: `%s`", snippet(expectedLine));
+					}
+					if (matches(expectedLine, actualDeque.peek())) {
+						continue main;
+					}
+					actualDeque.pop();
+				}
+			}
+
+			int number = expectedLines.size() - expectedDeque.size(); // 1-based line number
+			fail(expectedLines, actualLines, "expected line #%d:`%s` don't match", number, snippet(expectedLine));
+		}
+
+		// after math
+		if (!actualDeque.isEmpty()) {
+			fail(expectedLines, actualLines, "more actual lines than expected: %d", actualDeque.size());
+		}
+	}
+
+	private static String snippet(String line) {
+		if (line.length() <= MAX_SNIPPET_LENGTH) {
+			return line;
+		}
+		return line.substring(0, MAX_SNIPPET_LENGTH - 5) + "[...]";
+	}
+
+	private static void fail(List<String> expectedLines, List<String> actualLines, String format, Object... args) {
+		if (expectedLines.size() > MAX_SNIPPET_LENGTH) {
+			expectedLines.subList(0, MAX_SNIPPET_LENGTH);
+		}
+		if (actualLines.size() > MAX_SNIPPET_LENGTH) {
+			actualLines.subList(0, MAX_SNIPPET_LENGTH);
+		}
+		// use standard assertEquals(Object, Object, message) to let IDEs present the textual difference
+		String expected = String.join(System.lineSeparator(), expectedLines);
+		String actual = String.join(System.lineSeparator(), actualLines);
+		assertEquals(expected, actual, format(format, args));
+	}
+
+	static boolean isFastForwardLine(String line) {
+		line = line.trim();
+		return line.length() >= 4 && line.startsWith(">>") && line.endsWith(">>");
+	}
+
+	static int parseFastForwardLimit(String fastForwardLine) {
+		String text = fastForwardLine.trim().substring(2, fastForwardLine.length() - 2).trim();
+		try {
+			int limit = Integer.parseInt(text);
+			condition(limit > 0, "fast-forward limit must be greater than zero, it is: " + limit);
```

Comment:
Also, when building a string you are required to use the `Supplier<String>` variant of `condition()`.